include/seqan3/alphabet/composite/semialphabet_any.hpp

```diff
@@ -79,7 +79,7 @@ class semialphabet_any : public alphabet_base<semialphabet_any<size>, size, void
     //!\brief Enable conversion of semialphabet_any into other (semi-)alphabet of the same size
     template <semialphabet other_alph_t>
     //!\cond
-        requires (alphabet_size<other_alph_t> == size)
+        requires ((alphabet_size<other_alph_t> == size) && std::regular<other_alph_t>)
```

Comment:
Do you need only default-constructibility, or also equality comparison and copiability (=regular)? 
So either the requirement or the commit message is not correct :)

Reply:
`semialphabet` already includes everything from `std::regular` other than default-constructibility.